src/user.rs

```diff
@@ -141,17 +132,26 @@ impl User {
         log::debug!("user '{}' act => {}", self.localpart, "SYNC");
         match self.client.sync(user_notifier).await {
             SyncResult::Ok {
-                direct_messages,
+                rooms,
                 invited_rooms,
                 cancel_sync,
-                channels,
             } => {
                 log::debug!(
                     "user '{}' has {} dm rooms",
                     self.localpart,
-                    direct_messages.len()
+                    rooms
+                        .iter()
+                        .filter(|r| matches!(r.1, RoomType::DirectMessage))
```

Comment:
Extract function `rooms(rooms, RoomType)` to be used here and below at the least, or reuse `get_room_count` somehow.